spec/support/connection_monitoring_and_pooling/verifier.rb

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  module CMAP
+    class Verifier
+      include RSpec::Matchers
+
+      def initialize(test_instance)
+        @test_instance = test_instance
+      end
+
+      attr_reader :test_instance
+
+      # Compare the existing CMAP events data and the expected CMAP events.
+      #
+      # Uses RSpec matchers and raises expectation failures if there is a
+      # mismatch.
+      def verify_events(actual_events)
+        expected_events = test_instance.events
+        if expected_events.nil?
+          expect(actual_events).to be nil
+        elsif expected_events.empty?
+          expect(actual_events).to be_empty
+        else
+          expect(actual_events).not_to be nil
+          expected_events.each do |event|
+            verify_event(event, actual_events)
+          end
+          actual_events.each do |event|
+            verify_event(event, expected_events)
+          end
+        end
+      end
+
+      private
+
+      def verify_event(event, events)
+        expect(
+          events.any? { |other| verify_hashes(event, other) }
```

Comment:
Was this copied from another verifier? This should be a simple loop and verify_hashes should use assertions like verify_events does.

Reply:
I've revised the implementation of this part of the verifier; let me know if you want any additional changes